### PR TITLE
SSO Controller Architecture documentation

### PR DESCRIPTION
**Pull request by bastjan**

### CODE REVIEW

File: docs/modules/ROOT/pages/references/architecture/single_sign_on.adoc [hunk 1]
```diff
@@ -0,0 +1,215 @@
+= Single Sign On
+
+== Problem Statement
+
+Our current log in system, based on LDAP has security issues.
```
Comment (simu) [inline docs/modules/ROOT/pages/references/architecture/single_sign_on.adoc:5]: If you want a subordinate clause (Nebensatz), it should be something like

```suggestion
Our current log in system, which is based on LDAP, has security issues.
```

File: docs/modules/ROOT/pages/references/architecture/single_sign_on.adoc [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+It doesn't allow for 2FA, and we enter our password in a lot of different masks, that could in theory be compromised.
```
Comment (simu) [inline docs/modules/ROOT/pages/references/architecture/single_sign_on.adoc:6]:
```suggestion
It doesn't allow for 2FA, and we enter our password in a lot of different masks that could be compromised.
```

File: docs/modules/ROOT/pages/references/architecture/single_sign_on.adoc [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+To allow for a way to write back static "dynamic" facts, we add a ConfigMap to the Lieutenant namespace that gets added to the dynamic facts.
+This config map is managed by `component-steward` and new facts can be added through the hierarchy.
+
+[source,yaml]
+----
+parameters:
+  steward:
+    additionalFacts:
+      vshnLdapServiceId: "{vshnLdap:serviceId}"
```
Comment (simu) [inline docs/modules/ROOT/pages/references/architecture/single_sign_on.adoc:102]:
```suggestion
      vshnLdapServiceId: "${vshnLdap:serviceId}"
```

File: docs/modules/ROOT/pages/references/architecture/single_sign_on.adoc [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  facts:
+    kubernetesVersion: '{"buildDate":"2023-09-11T02:22:18Z","compiler":"gc","gitCommit":"f10a517f7199bdae922a70893d85eb96a76f5c2d","gitTreeState":"clean","gitVersion":"v1.26.7+c7ee51f","goVersion":"go1.19.10
+      X:strictfipsruntime","major":"1","minor":"26","platform":"linux/amd64"}'
+    openshiftVersion: '{"Major":"4","Minor":"13","Patch":"13"}'
+    oauthDomain: https://oauth-openshift.apps.c-holy-fire-9875.dev <6>
+    vshnLdapServiceId: ClusterHolyFire9875 <7>
+----
+<1> The `sso.appuio.io/keycloak-client` finalizer is added to the cluster object to allow cleanup of the Keycloak client when the cluster is deleted.
+<2> The `oidc.sso.appuio.io/redirect-uris` annotation is used to override the default redirect uris.
+<3> The `sso.appuio.io/vshn-ldap-id` annotation is used to override the default LDAP group mapping.
```
Comment (simu) [inline docs/modules/ROOT/pages/references/architecture/single_sign_on.adoc:166]: There's an argument here that `sso.appuio.io` isn't the right domain for any of the annotations.

I'd argue that the SSO controller itself is useful for Project Syn in general, and therefore the first two annotations could be prefixed with `sso.syn.tools` and `oidc.sso.syn.tools`, while the last annotation is so VSHN-specific that it really should be `sso.vshn.net/ldap-id` or similar.

Comment (bastjan) [reply docs/modules/ROOT/pages/references/architecture/single_sign_on.adoc]: True. Much happier with your proposed naming.

**Review by simu — APPROVED** [review]
LGTM now

**Pull request merged**